service/src/test/java/greencity/service/ubs/UBSManagementServiceImplTest.java

```diff
@@ -2886,4 +2886,53 @@ void saveOrderIdForRefundThrowsNotFoundExceptionTest() {
         assertThrows(NotFoundException.class, () -> ubsManagementService.getNotTakenOrderReason(1L));
         verify(orderRepository).findById(1L);
     }
+
+    @Test
+    void updateOrderAdminPageInfoAndSaveReasonTest() {
+        var dto = updateOrderPageAdminDto();
+        MockMultipartFile[] multipartFiles = new MockMultipartFile[0];
+
+        Order order = getOrder();
+
+        TariffsInfo tariffsInfo = getTariffsInfo();
+        order.setOrderDate(LocalDateTime.now()).setTariffsInfo(tariffsInfo);
+
+        Employee employee = getEmployee();
+
+        when(orderRepository.findById(1L)).thenReturn(Optional.of(order));
+        when(employeeRepository.findByEmail("test@gmail.com")).thenReturn(Optional.of(employee));
+        when(tariffsInfoRepository.findTariffsInfoByIdForEmployee(anyLong(), anyLong()))
+            .thenReturn(Optional.of(tariffsInfo));
+        when(paymentRepository.findAllByOrderId(1L)).thenReturn(List.of(getPayment()));
+        when(ubsClientService.updateUbsUserInfoInOrder(ModelUtils.getUbsCustomersDtoUpdate(),
+            "test@gmail.com")).thenReturn(ModelUtils.getUbsCustomersDto());
+
+        UpdateOrderPageAdminDto updateOrderPageAdminDto = updateOrderPageAdminDto();
+        updateOrderPageAdminDto.setUserInfoDto(ModelUtils.getUbsCustomersDtoUpdate());
+
+        when(orderAddressRepository.findById(1L)).thenReturn(Optional.of(getOrderAddress()));
+        when(receivingStationRepository.findAll()).thenReturn(List.of(getReceivingStation()));
+
+        var receivingStation = getReceivingStation();
+
+        when(receivingStationRepository.findById(1L)).thenReturn(Optional.of(receivingStation));
+        when(orderRepository.getOrderDetails(anyLong())).thenReturn(Optional.ofNullable(getOrdersStatusFormedDto()));
+        when(bagRepository.findById(1)).thenReturn(Optional.of(ModelUtils.getTariffBag()));
+
+        ubsManagementService.updateOrderAdminPageInfoAndSaveReason(1L, dto, "en", "test@gmail.com", "desc",
+            multipartFiles);
+
+        verify(orderRepository, times(9)).findById(1L);
```

Comment:
Why so much invoking, can we invoke this once?

Reply:
fixed